docs/topics/guideline/solidity.md

```diff
@@ -0,0 +1,307 @@
+
+Internal solve the testing issue. A mock child contract can be used to access the variable.
+
+Public solve all the issues the only downside is that it adds a default getter and calling a public method can make the
+contract code and gas consumption higher.
+
+### TO DISCUSS: Use of private view methods
+
+Usually view methods are used as a getters or to do some reusable calculation. Exposing the method helps to test the
+contract and is useful to check the internal state of a deployed contract. Methods can be exposed using public or
+external. If possible prefer external because external function's parameters are not copied into memory but are read
+directly from `calldata` and this can save some gas.
+
+### Calling internal functions is cheaper
+
+Calling internal functions is cheaper than calling public functions. When you call a public function all the parameters
+are copied again into memory and passed to that function. By contrast, when you call an internal function, references of
+those parameters are passed, and they are not copied again.
+
+## Others
+
+### Lock pragmas to specific compiler version
+
+Pragma must select a specific version of the compiler, instead of using `pragma solidity ^0.8.2;`
+use `pragma solidity 0.8.2;`.
+
+### Use interface type instead of the address for type safety
+
+```
+interface SomethingToCall {
+    function call() returns (bool);
+}
+
+contract Example {
+    function Good(SomethingToCall _something) internal returns (bool) {
+    }
+
+    function Bad(address _addr) internal returns (bool) {
+        SomethingToCall something = SomethingToCall(_addr);
+    }
+}
+```
+
+### Use short reason strings
+
+The solidity linter warns about it. Reason strings (those used in `revert`) that takes more than 32 bytes make the
+contract byte code bigger.
+
+### TO DISCUSS: Pull over Push
+
+From the list of [Solidity Patterns](https://github.com/fravoll/solidity-patterns) this one is a very important one: "
+Sending ether to another address in Ethereum involves a call to the receiving entity".
+
+### Use of view methods that reverts
+
+Usually view methods return some value. Is better to just return a value indicating a failure than doing a revert that
+is reserved for state changes and is hard to catch with some blockchain client libraries. Assert can be used to test for
+internal errors, and to check invariants.
+
+## Packing
+
+Storage variables can be packed. Pack your variables by declaring them in the right order. Tight packing variables in
+structs can be useful to make the code
+clear: [Tight Variable Packing](https://fravoll.github.io/solidity-patterns/tight_variable_packing.html)
+
+For some calculations take into account that uint8 used alone can be more expensive than uint256 because the EVM word is
+32bytes/256bits.
+
+## Mapping are cheaper than arrays
+
+Mapping are cheaper than arrays and a lot of times using a customization (solidity doesn't support generics)
```

Comment:
cheaper in what (storage, computation) Do you have a doc stating that?

Reply:
I didn't find a benchmark. Probably the title is too general and it depends on the use case.
What I remember from the top of my head is that it was related to boundary checking but I may be wrong.
A google search gives you: https://ethereum.stackexchange.com/questions/37549/array-or-mapping-which-costs-more-gas/37594 
Maybe we can remove this one.

Reply:
Removed.